backend/src/utils/resourceUtils.ts

```diff
@@ -80,58 +76,82 @@ const fetchInstalledKfdefs = async (fastify: KubeFastifyInstance): Promise<KfDef
   return kfdef?.spec?.applications || [];
 };
 
-const fetchApplicationDefs = (): Promise<ODHApp[]> => {
-  const normalizedPath = path.join(__dirname, '../../../data/applications');
-  const applicationDefs: ODHApp[] = [];
+const fetchRhodsApplications = async (
+  fastify: KubeFastifyInstance,
+): Promise<RhodsApplication[]> => {
+  const customObjectsApi = fastify.kube.customObjectsApi;
+  const namespace = fastify.kube.namespace;
   const featureFlags = getComponentFeatureFlags();
-  fs.readdirSync(normalizedPath).forEach((file) => {
-    if (yamlRegExp.test(file)) {
-      try {
-        const doc = jsYaml.load(fs.readFileSync(path.join(normalizedPath, file), 'utf8'));
-        if (!doc.spec.featureFlag || featureFlags[doc.spec.featureFlag]) {
-          applicationDefs.push(doc);
+
+  let rhodsApplications: RhodsApplication[];
+  try {
+    const res = await customObjectsApi.listNamespacedCustomObject(
+      'applications.console.openshift.io',
+      'v1alpha1',
+      namespace,
+      'rhodsapplications',
+    );
+    const cas = (res?.body as { items: RhodsApplication[] })?.items;
+    if (cas?.length) {
```

Comment:
Don't check length here or initialize `rhodsApplications` to []. This function should not return `undefined`.

Reply:
done